packages/cli/src/utils/getProjects.ts

```diff
@@ -1,55 +1,120 @@
-  const projects: (Project | undefined)[] = []
-
-  for (const rootFileName of await fs.readdir("./")) {
-    const rootFile = path.join(".", rootFileName)
-    if ((await fs.stat(rootFile)).isDirectory()) {
-      for (const nestedFileName of await fs.readdir(rootFile)) {
-        const nestedFile = path.join(rootFile, nestedFileName)
-        if ((await fs.stat(nestedFile)).isDirectory()) {
-          // Deeper projects not supported atm
-        } else {
-          projects.push(await getProject(nestedFile))
-        }
+  const projects = await getProjectsInternal()
+  return projects.sort((a, b) => (a.type === "client" ? -1 : 1))
+}
+
+async function getProjectsInternal(): Promise<Project[]> {
+  const projects: Project[] = []
+
+  // Get projects based on workspace-config
+  const workspacePaths = await getWorkspacePaths(".")
+  if (workspacePaths.length > 0) {
+    for (const workspacePath of workspacePaths) {
+      const project = await getProject(workspacePath)
+      if (project) {
+        projects.push(project)
+      }
+    }
+  }
+  if (projects.length > 0) {
+    return projects
+  }
+
+  // Get nested projects (1 level deep)
+  for (const fileName of await fs.readdir(".")) {
+    const filePath = path.join(".", fileName)
+    if ((await fs.stat(filePath)).isDirectory()) {
+      const nestedProject = await getProject(filePath)
+      if (nestedProject) {
+        projects.push(nestedProject)
       }
-    } else {
-      projects.push(await getProject(rootFile))
     }
   }
+  if (projects.length > 0) {
+    return projects
+  }
 
+  // Get project at the root level
+  const rootProject = await getProject(".")
+  if (rootProject) {
+    projects.push(rootProject)
+  }
   return projects
-    .filter((p): p is Project => !!p)
-    .sort((a, b) => (a.type === "client" ? -1 : 1))
 }
 
-async function getProject(filePath: string): Promise<Project | undefined> {
-  if (!filePath.endsWith("package.json")) {
+interface PackageJson {
+  workspaces?: string[]
+  dependencies?: { [key: string]: string }
+  devDependencies?: { [key: string]: string }
+}
+
+async function getPackageJson(
+  dirPath: string,
+): Promise<PackageJson | undefined> {
+  const packagePath = path.join(dirPath, "package.json")
+  if (await exists(packagePath)) {
+    const packageString = await fs.readFile(packagePath, "utf-8")
+    return JSON.parse(packageString)
+  } else {
     return undefined
   }
+}
+
+async function getWorkspacePaths(dirPath: string): Promise<string[]> {
+  const packageJson = await getPackageJson(dirPath)
+  const result: string[] = []
+
+  if (packageJson?.workspaces) {
+    for (const pattern of packageJson.workspaces) {
+      result.push(...(await getWorkspacePathsForGlobPattern(pattern)))
+    }
+  }
 
-  const packageString = await fs.readFile(filePath, "utf-8")
-  const packageJson = JSON.parse(packageString)
+  return result
+}
+
+async function getWorkspacePathsForGlobPattern(
```

Comment:
Ah, didn't spot the sync-alternative. Let's keep it like this, because we're probably going to make it all async somewhere in the future anyway, right?